anyway/views/news_flash/api.py

```diff
@@ -199,19 +200,75 @@ def gen_news_flash_query(
 
     return query
 
+def filter_by_road_number(query: Query, value: float, **kwargs) -> Query:
+    return query.filter(NewsFlash.road1 == value)
+
+def filter_by_source(query: Query, value: Optional[List[BE_CONST.Source]], **kwargs) -> Query:
+    if value is None:
+        return query
+    sources = set([source.value for source in value])
+    critical = BE_CONST.Source.CRITICAL.value
+    if critical in sources:
+        query = query.filter(NewsFlash.critical==True)
+        sources.remove(critical)
+        if len(sources) == 0:
+            return query
```

Comment:
Hi, I think it is wrong to relate to a specific field as if it is a source. This is wrong modeling. Sources has different behavior than a specific field. Such code will require many locations of specific logic, will be hard to maintain and prone to errors.